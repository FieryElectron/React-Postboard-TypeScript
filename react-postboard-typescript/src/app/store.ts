import { configureStore, ThunkAction, Action } from '@reduxjs/toolkit';

import signInPageReducer from '../components/SignInPage/signInPageSlice';
import postBoardPageReducer from '../components/PostBoardPage/postBoardPageSlice';

export const store = configureStore({
  reducer: {
    signInPage: signInPageReducer,
    postBoardPage: postBoardPageReducer,
  },
});

export type AppDispatch = typeof store.dispatch;
export type RootState = ReturnType<typeof store.getState>;
export type AppThunk<ReturnType = void> = ThunkAction<
  ReturnType,
  RootState,
  unknown,
  Action<string>
>;
