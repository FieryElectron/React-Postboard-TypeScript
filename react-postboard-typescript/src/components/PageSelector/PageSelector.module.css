.pageSelectorTitle {
    font-size: 20px;
    font-weight: 500;
    margin-right: 5px;
}

.pageSelectorSelect {
    background-color: #0563af;
    color: white;
    padding: 5px;
    width: 50px;
    border: none;
    font-size: 15px;
    outline: none;
    border-radius: 5px;
}

.pageSelectorContainer {
    display: flex;
    justify-content: center;
    align-items: center;
}