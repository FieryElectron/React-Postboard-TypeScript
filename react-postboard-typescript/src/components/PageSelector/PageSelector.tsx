import styles from './PageSelector.module.css';

import { useAppSelector, useAppDispatch } from '../../app/hooks';
import { useNavigate } from 'react-router-dom';

import {
    selectPages,
    setSelectedPage,
    getPosts,
} from '../PostBoardPage/postBoardPageSlice';

const PageSelector = () => {

    const dispatch = useAppDispatch();
    const navigate = useNavigate();

    const pages = useAppSelector(selectPages);

    const selectedPageOnChange = (e: React.ChangeEvent<HTMLSelectElement>) => {
        const page = parseInt(e.target.value);

        dispatch(setSelectedPage(page));
        dispatch(getPosts(navigate));
    }

    return (
        <div data-testid='pageSelectorContainer' className={styles.pageSelectorContainer}>
            <span data-testid='pageSelectorTitle' className={styles.pageSelectorTitle}>Page</span>
            <select id='pageSelectorId' data-testid='pageSelectorSelect' className={styles.pageSelectorSelect} onChange={selectedPageOnChange}>
                {pages.map((value ,index,list) => <option key={value} value={value} >{value}</option>)}
            </select>
        </div>
    )
}

export default PageSelector;