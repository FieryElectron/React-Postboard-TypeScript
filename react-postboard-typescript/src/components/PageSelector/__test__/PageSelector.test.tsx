import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { act } from 'react-dom/test-utils';
import '@testing-library/jest-dom/extend-expect';
import PageSelector from '../PageSelector';
import { Provider } from 'react-redux';
import { BrowserRouter as Router} from 'react-router-dom'
import { store } from '../../../app/store';

let container: HTMLElement;

beforeEach(() => {
  container = document.createElement('div');
  document.body.appendChild(container);
});

afterEach(() => {
  document.body.removeChild(container);
  container.remove();
});

it('empty test', () => {

});

it('renders PageSelector without crashing', () => {
  act(() => {
    render(<Provider store={store}><Router><PageSelector /></Router></Provider>);
  });
});

it('renders PageSelector class correctly', () => {
  act(() => {
    render(<Provider store={store}><Router><PageSelector /></Router></Provider>);
  });

  const pageSelectorContainer = document.querySelector('[data-testid=pageSelectorContainer]')!;
  const pageSelectorTitle = document.querySelector('[data-testid=pageSelectorTitle]')!;
  const pageSelectorSelect = document.querySelector('[data-testid=pageSelectorSelect]')!;

  expect(pageSelectorContainer).toHaveAttribute('class', 'pageSelectorContainer');
  expect(pageSelectorTitle).toHaveAttribute('class', 'pageSelectorTitle');
  expect(pageSelectorSelect).toHaveAttribute('class', 'pageSelectorSelect');
});


it('matchs snapshot 1', ()=>{
    const tree = renderer.create(<Provider store={store}><Router><PageSelector/></Router></Provider>).toJSON();
    expect(tree).toMatchSnapshot();
}); 
 



