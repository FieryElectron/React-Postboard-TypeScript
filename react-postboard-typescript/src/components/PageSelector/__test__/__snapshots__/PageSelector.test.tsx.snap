// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matchs snapshot 1 1`] = `
<div
  className="pageSelectorContainer"
  data-testid="pageSelectorContainer"
>
  <span
    className="pageSelectorTitle"
    data-testid="pageSelectorTitle"
  >
    Page
  </span>
  <select
    className="pageSelectorSelect"
    data-testid="pageSelectorSelect"
    id="pageSelectorId"
    onChange={[Function]}
  >
    <option
      value={1}
    >
      1
    </option>
    <option
      value={2}
    >
      2
    </option>
    <option
      value={3}
    >
      3
    </option>
    <option
      value={4}
    >
      4
    </option>
    <option
      value={5}
    >
      5
    </option>
  </select>
</div>
`;
